subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,810 @@
+
+++++
+<figure>
+    <title>Processing non-source files for a source set</title>
+    <imageobject>
+        <imagedata fileref="img/java-sourcesets-process-resources.png" width="170mm"/>
+    </imageobject>
+</figure>
+++++
+
+As before, the shaded boxes represent properties of the source set, which in this case comprises the locations of the resource files and where they are copied to.
+
+In addition to the `main` source set, the Java Plugin defines a `test` source set that represents the project's tests. This source set is used by the `test` task, which runs the tests using https://junit.org/[JUnit] or http://testng.org/[TestNG].
+
+Projects typically use this source set for unit tests, but you can use it for integration, acceptance and other types of test if you wish. That said, most projects <<sec:custom_java_source_sets,define new source sets>> for those other test types because they require special setup or classpaths.
+
+You'll learn more about source sets and the features they provide in other sections of the chapter as many facets of building a Java project make use of them.
+
+[[sec:java_dependency_management_overview]]
+=== Managing your dependencies 
+
+The vast majority of Java projects rely on libraries, so managing a project's dependencies is an important part of building a Java project. Dependency management is a big topic, so we will focus on the basics for Java projects here. If you'd like to dive into the detail, check out the <<introduction_dependency_management,introduction to dependency management>>.
+
+Specifying the dependencies for your Java project requires just three pieces of information:
+
+ * Which dependency you need, such as a name and version
+ * What it's needed for, e.g. compilation or running
+ * Where to look for it
+
+The first two are specified in a `dependencies {}` block and the second in a `repositories {}` block. For example, to tell Gradle that your project requires version 3.6.7 of http://hibernate.org/[Hibernate] Core to compile and run your production code, and that you want to download the library from the Maven Central repository, you can use the following fragment:
+
+++++
+<sample id="javaPluginDependencyMgmt" dir="userguide/java/basic" title="Declaring dependencies">
+    <sourcefile file="build.gradle" snippet="java-dependency-mgmt"/>
+</sample>
+++++
+
+The Gradle terminology for the three elements is as follows:
+
+ * _Repository_ — where to look for dependencies (`mavenCentral()`)
+ * _Configuration_ - how the dependency will be used (`implementation`) — a more flexible form of Maven scopes
+ * _Dependency coordinate_ — the ID of the dependency, usually in the form '__<group>__:__<module>__:__<version>__' (or '__<groupId>__:__<artifactId>__:__<version>__' in Maven terminology)
+
+As far as configurations go, the main ones of interest are:
```

Comment:
Should also list and explain `api`.

Reply:
But that only exists for the Java Library Plugin. I'll review. Maybe it makes sense to call it out here as well as in the _Building Java libraries_ section.

Reply:
I've called it out just after the list as I want it to be absolutely clear that it's not one of the configurations added by the Java Plugin.

Reply:
For reference: https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java#L116. Though I'd have to admit that it is not fully clear to me when you would use `api` in a Java standalone application. 

@gradle/dependency-management Can you comment on this?

Reply:
The name is defined there, but the configuration is not added by the Java Plugin. Strangely, at first glance at least, the plugin does add the `apiElements` configuration.